Eastern-Research-Group/mywaterway
# Feature/573 characteristics filter

## Related Issues:
* [HMW-573](https://jira.epa.gov/browse/HMW-573)

## Main Changes:
* Added a multi-selectable characteristic filter to the _Past Water Conditions_ accordion lists. This input's options are drawn from the WQP `periodOfRecord` summary service, so is not immediately active.
* Moved the point of fetching `periodOfRecord` data up into the `useMonitoringLocationsLayers` hook. There is an option to disable, which is used on the Monitoring Report page. While the service may still be used on that page in [HMW-572](https://jira.epa.gov/browse/HMW-572), it will likely be on an as-needed basis.
* Updated some state items to use sensible defaults instead of `null` (much more feasible now due to the amount of TypeScript).

## Steps To Test:
1. Navigate to http://localhost:3000/community/dc/monitoring.
2. In the _Past Water Conditions_ tab, select a characteristic from the **Filter by Characteristic** input, and confirm the change is reflected in the list.
3. Go to the Monitoring Report page of several of those list items and confirm the selected characteristic is indeed present.
4. On the Community page, ensure that the date slider continues to filter locations alongside the characteristic filter.
5. Go to the _Water Monitoring Locations_ section of the Overview tab.
6. Confirm the characteristic filter is present.
7. Toggle of the Past Water Conditions, then confirm the characteristic filter is no longer visible.
8. Go to http://localhost:3000/community/Portland,%20OR,%20USA/monitoring, and verify the toggle table and list items are still populated before the date slider and characteristic input are ready.
9. Confirm the filter works properly for the list view on tribal pages, e.g. http://localhost:3000/tribe/UTEMTN.
